api/src/main/java/org/apache/iceberg/ContentFile.java

```diff
@@ -113,6 +113,16 @@ default Integer sortOrderId() {
     return null;
   }
 
+  /** Returns the data sequence number of the snapshot in which the file should be applied. */
```

Comment:
What about something like this?

```
/**
 * Returns the data sequence number of the file.
 *
 * <p>This method represents the sequence number to which the file should apply. Note the data
 * sequence number may differ from the sequence number of the snapshot in which the underlying
 * file was added. New snapshots can add files that belong to older sequence numbers (e.g.
 * compaction). The data sequence number also does not change when the file is marked as deleted.
 *
 * <p>This method can return null if the data sequence number is unknown. This may happen while
 * reading a v2 manifest that did not persist the data sequence number for manifest entries with
 * status DELETED (older Iceberg versions).
 */
```

Reply:
@gaborkaszab did you push the change? I am still seeing the old Javadoc.